.github/workflows/test-php.yml

```diff
@@ -12,24 +12,42 @@ on:
 jobs:
   test-php:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php: ['8.1']
+        composer-mode: ['low-deps', 'high-deps']
 
     steps:
       - uses: actions/checkout@v3
 
-      - name: Cache Composer dependencies
-        uses: actions/cache@v3
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
         with:
-          path: /tmp/composer-cache
-          key: ubuntu-latest-${{ hashFiles('php/composer.lock') }}
+          php-version: "${{ matrix.php }}"
 
-      - name: Set up PHP
-        uses: php-actions/composer@v6
+      - name: Discover composer cache directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-dir)"
+        working-directory: php
+
+      - name: Cache composer
+        uses: actions/cache@v3
         with:
-          working_dir: php
+          path: "${{ steps.composer-cache.outputs.dir }}"
+          key: composer
```

Comment:
I've not done anything more complex here because composer doesn't evict unused stuff from the cache so AFAIK we'll just end up with a large cache of ZIPs of all the different versions of dependencies